CMakeLists.txt

```diff
@@ -212,8 +212,19 @@ endif()
 
 if(NOT NONET)
   if(DEVILUTIONX_SYSTEM_LIBSODIUM)
-    set(sodium_USE_STATIC_LIBS ${DEVILUTIONX_STATIC_LIBSODIUM})
-    find_package(sodium REQUIRED)
+    # We really need to use pkgconfig to find libsodium.
+    # They don't seem to support the CMake module method anymore and
+    # pkgconfig is simpler anyway.
+    # https://github.com/diasurgical/devilutionX/issues/2615
+    find_package(PkgConfig)
+    if(PkgConfig_FOUND)

```

Comment:
This doesn't make sense in the main CMakeLists.txt. Findsodium.cmake should be adjusted instead.
As far as I can tell, it already tries pkgconfig first but perhaps it does so incorrectly.

Reply:
Note that I mean our own Findsodium.cmake here: https://github.com/diasurgical/devilutionX/blob/master/CMake/Findsodium.cmake

We do not use the upstream (now deleted) Findsodium.cmake

Reply:
Thank for the pointers! On it.